### PR TITLE
feat: Added a function for finding the least common multiple

### PR DESCRIPTION
**Pull request by tjgurwara99**
#### Description of Change
I have added a function that outputs the least common multiple of two numbers.

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTION.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTION.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] Sort by alphabetical order
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: Function to find the least common multiple of two numbers.

### CODE REVIEW

File: math/least_common_multiple.cpp [hunk 1]
```diff
@@ -0,0 +1,41 @@
+// Copyright 2020 @author tjgurwara99
```
Comment (Panquesito7) [inline math/least_common_multiple.cpp:1]: Please change this to a comment structure like:

```c
/**
 * Copyright 2020 @author tjgurwara99
 * @file
 *
 * (basic description about the program here)
```

File: math/least_common_multiple.cpp [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    return gcd(x, y - temp * x);
+}
+
+/*
+ * Now we use the fact that lcm(x,y) * gcd(x,y)  = x * y
+ */
+unsigned int lcm(unsigned int x, unsigned int y) { return x * y / gcd(x, y); }
+
+int main() {
+    int ans = lcm(5, 10);
```
Comment (ayaankhan98) [inline math/least_common_multiple.cpp:38]: i suggest you to write some test cases to validate the algorithm properly, please add a function `void tests()` and write some tests in this functions using `assert` statements and call this function from `main()`

File: math/least_common_multiple.cpp [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+    int temp = y / x;
+    return gcd(x, y - temp * x);
+}
+
+/*
+ * Now we use the fact that lcm(x,y) * gcd(x,y)  = x * y
+ */
+unsigned int lcm(unsigned int x, unsigned int y) { return x * y / gcd(x, y); }
+
```
[the remark this answers is not in the document]
Comment (tjgurwara99) [reply math/least_common_multiple.cpp:44]: Done

**Comment by tjgurwara99** [comment]
I have taken onboard all the comments and tried to fix the issues, please take a look now.

**Comment by ayaankhan98** [comment]
@tjgurwara99 have a look at #807 files are very well documented  in this PR, please do the documentaion in similar manner.
Thanks! for your contribution

File: math/least_common_multiple.cpp [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+#include <cassert>
+#include <iostream>
+
+/**
+ * Function for finding greatest common divisor of two numbers.
+ * @params two integers x and y whose gcd we want to find.
+ * @return greatest common divisor of x and y.
+ */
+unsigned int gcd(unsigned int x, unsigned int y) {
```
Comment (kvedala) [inline math/least_common_multiple.cpp:16]: Could you make another pull request in which the functions are templated?

Eg, `gcd` and `lcm` Function inputs can be int, long, long long, signed or unsigned.

```cpp
Template<typename T>
gcd(const T &a, const T &b)
{....}
```

Because it is a recursive function, use of const references gives a faster memory access. The templates

File: math/least_common_multiple.cpp [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+ *
+ * A basic implementation of LCM function
+ */
+
+#include <cassert>
+#include <iostream>
+
+/**
+ * Function for finding greatest common divisor of two numbers.
+ * @params two integers x and y whose gcd we want to find.
```
Comment (kvedala) [inline math/least_common_multiple.cpp:13]: The syntax is incorrect please refer #807 and [this link](https://www.doxygen.nl/manual/docblocks.html)

File: math/least_common_multiple.cpp [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    // Again the following is valid because we have checked whether x == 0
+
+    int temp = y / x;
+    return gcd(x, y - temp * x);
+}
+
+/**
+ * Function for finding the least common multiple of two numbers.
+ * @params integer x and y whose lcm we want to find.
+ * @return lcm of x and y using the relation x * y = gcd(x, y) * lcm(x, y)
```
Comment (kvedala) [inline math/least_common_multiple.cpp:41]: You can use LaTeX syntax for math formulae in the documentation:
```Latex
\f$x \cdot y = \mathrm{gcd}(x, y) \cdot /mathrm{lcm}(x, y)\f$
```
[Example output](https://kvedala.github.io/C-Plus-Plus/d7/d6a/bisection__method_8cpp.html#details)

File: math/least_common_multiple.cpp (hunk 4 above)
Comment (tjgurwara99) [reply math/least_common_multiple.cpp:16]: @kvedala I have opened another pull request #845. Once that is complete I'll create a new pull request for these changes.

File: math/least_common_multiple.cpp (hunk 6 above)
Comment (tjgurwara99) [reply math/least_common_multiple.cpp:41]: I didn't know about this feature of Doxygen, thanks for pointing it out!

**Comment by kvedala** [comment]
@tjgurwara99 This is the autogenerated documentation for this file -> [link](https://kvedala.github.io/C-Plus-Plus/d4/d21/least__common__multiple_8cpp.html). Note:
1. the correct tag is `param` and not `params`. hence, all the parameter tags got ignored
2. no `\brief` tag in the beginning of the file did not create any summary in the [list of files page](https://kvedala.github.io/C-Plus-Plus/files.html)
Thought the points would help you going forward and this was another point I realized should keep an eye while reviewing 😄

**Comment by tjgurwara99** [comment]
> @tjgurwara99 This is the autogenerated documentation for this file -> [link](https://kvedala.github.io/C-Plus-Plus/d4/d21/least__common__multiple_8cpp.html). Note:
> 
> 1. the correct tag is `param` and not `params`. hence, all the parameter tags got ignored
> 2. no `\brief` tag in the beginning of the file did not create any summary in the [list of files page](https://kvedala.github.io/C-Plus-Plus/files.html)
>    Thought the points would help you going forward and this was another point I realized should keep an eye while reviewing 😄

@kvedala Could you show me the config file (cmake or Doxyfile) you use for Doxygen, I want to try and add that to other projects but don't seem to be able to do it.

**Comment by kvedala** [comment]
> > @tjgurwara99 This is the autogenerated documentation for this file -> [link](https://kvedala.github.io/C-Plus-Plus/d4/d21/least__common__multiple_8cpp.html). Note:
> > 
> > 1. the correct tag is `param` and not `params`. hence, all the parameter tags got ignored
> > 2. no `\brief` tag in the beginning of the file did not create any summary in the [list of files page](https://kvedala.github.io/C-Plus-Plus/files.html)
> >    Thought the points would help you going forward and this was another point I realized should keep an eye while reviewing 😄
> 
> @kvedala Could you show me the config file (cmake or Doxyfile) you use for Doxygen, I want to try and add that to other projects but don't seem to be able to do it.

In my setup, the doxygen gets configured by cmake itself. -> [right here](https://github.com/kvedala/C-Plus-Plus/blob/45e6f3c96de6a3f0edc610e05043c0f275d62d66/CMakeLists.txt#L21)
